### ISSUE TITLE
File.GetLastWriteTime throws on missing files on Linux, but not Windows

### ISSUE THREAD

**Issue by AArnott** [dotnet/corefx#17368]
On .NET Framework and CoreCLR on Windows, I can safely call `File.GetLastWriteTime` on a file that does not exist, and I get "December 31, 1600 4:00:00 PM" returned. 

But on Linux, CoreCLR throws `FileNotFoundException`.

```
System.IO.FileNotFoundException: Could not find file '/mnt/c/git/libgit2sharp/LibGit2Sharp/obj/Debug/netstandard1.3/AfterRebaseStepInfo.348465dd.generated.cs'. [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
File name: '/mnt/c/git/libgit2sharp/LibGit2Sharp/obj/Debug/netstandard1.3/AfterRebaseStepInfo.348465dd.generated.cs' [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
   at System.IO.FileSystemInfo.EnsureStatInitialized() [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
   at System.IO.FileSystemInfo.System.IO.IFileSystemObject.get_LastWriteTime() [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
   at System.IO.FileSystemInfo.get_LastWriteTimeUtc() [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
   at System.IO.UnixFileSystem.GetLastWriteTime(String fullPath) [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
   at System.IO.File.GetLastWriteTime(String path) [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
   at CodeGeneration.Roslyn.Tasks.Helper.<<Execute>b__38_1>d.MoveNext() [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
--- End of stack trace from previous location where exception was thrown --- [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
   at CodeGeneration.Roslyn.Tasks.Helper.Execute() [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
   at CallSite.Target(Closure , CallSite , Object ) [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
   at CodeGeneration.Roslyn.Tasks.GenerateCodeFromAttributes.Execute() [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__25.MoveNext() [/mnt/c/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
```

I notice that the problem isn't in `File.GetLastWriteTime` itself, because `FileSystemInfo.get_LastWriteTimeUtc()` also is not expected to throw in this case either.

I suggest for ease of interoperability, we keep corefx APIs behaving the same way across operating systems for this.

**Comment by danmosemsft** [comment]
Seems it's been that way since the start, and we even have a test to make sure

https://github.com/dotnet/corefx/blob/6451bed4d4a667d9c05a41c0843504d020d283ab/src/System.IO.FileSystem/tests/File/GetSetTimes.cs#L154

@ianhays know why?

**Comment by ianhays** [comment]
We've been getting a lot of these lately. Seems like votes in favor of revisiting the behavior. This is a duplicate of #1728, so lets move the discussion there.

I took this to a design review a while back and the sentiment was fairly strong that we should throw on Unix. I doubt that it will go any differently now, particularly since we already shipped with that behavior for 1.0. Of course if there is strong opinion contrary to that, then I'm certainly open to it.

Copying the last comment from #1728 for context:
>Design review feedback leaned towards allowing Unix to have a different behavior than Windows because of the Windows-specific notions of FromFileTime, though it wasn't entirely conclusive. Changing the Windows behavior is not what we want to do because of the breaking change it would cause, so the best option is different behaviors on different systems.

> Returning the windows 0 time on a Unix machine wouldn't make any sense and returning the Unix 0 time would still be inconsistent with the Windows 0 time, so the better alternative of throwing an exception is chosen.
